x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section.tsx

```diff
@@ -0,0 +1,510 @@
+import { i18n } from '@kbn/i18n';
+import { EuiPanel } from '@elastic/eui';
+import { EuiTitle } from '@elastic/eui';
+import { EuiIconTip } from '@elastic/eui';
+import { TopAlert } from '@kbn/observability-plugin/public/pages/alerts';
+import {
+  SERVICE_NAME,
+  TRANSACTION_TYPE,
+  ALERT_DURATION,
+  ALERT_END,
+} from '@kbn/rule-data-utils';
+import moment from 'moment';
+import { getOrRedirectToTransactionType } from '../../../context/apm_service/apm_service_context';
+import { useServiceAgentFetcher } from '../../../context/apm_service/use_service_agent_fetcher';
+import { useServiceTransactionTypesFetcher } from '../../../context/apm_service/use_service_transaction_types_fetcher';
+import { asPercent } from '../../../../common/utils/formatters';
+import { APIReturnType } from '../../../services/rest/create_call_apm_api';
+import { getDurationFormatter } from '../../../../common/utils/formatters/duration';
+import { ApmMlDetectorType } from '../../../../common/anomaly_detection/apm_ml_detectors';
+import { LatencyAggregationType } from '../../../../common/latency_aggregation_types';
+import { useFetcher } from '../../../hooks/use_fetcher';
+import { useTimeRange } from '../../../hooks/use_time_range';
+import { isTimeComparison } from '../../shared/time_comparison/get_comparison_options';
+import { getComparisonChartTheme } from '../../shared/time_comparison/get_comparison_chart_theme';
+import { getLatencyChartSelector } from '../../../selectors/latency_chart_selectors';
+import { TimeseriesChart } from '../../shared/charts/timeseries_chart';
+import { filterNil } from '../../shared/charts/latency_chart';
+import { usePreferredServiceAnomalyTimeseries } from '../../../hooks/use_preferred_service_anomaly_timeseries';
+import {
+  getMaxY,
+  getResponseTimeTickFormatter,
+} from '../../shared/charts/transaction_charts/helper';
+import { ChartPointerEventContextProvider } from '../../../context/chart_pointer_event/chart_pointer_event_context';
+import {
+  ChartType,
+  getTimeSeriesColor,
+} from '../../shared/charts/helper/get_timeseries_color';
+
+export interface AlertDetailsAppSectionProps {
+  rule: Rule<RuleTypeParams>;
+  alert: TopAlert;
+  timeZone: string;
+}
+
+const getAggsTypeFromRule = (ruleAggType: string): LatencyAggregationType => {
+  if (ruleAggType === '95th') return LatencyAggregationType.p95;
+  if (ruleAggType === '99th') return LatencyAggregationType.p99;
+  return LatencyAggregationType.avg;
+};
+export function AlertDetailsAppSectionTransactionDuration({
+  rule,
+  alert,
+  timeZone,
+}: AlertDetailsAppSectionProps) {
+  const params = rule.params;
+  const environment = String(params.environment);
+  const latencyAggregationType = getAggsTypeFromRule(
+    params.aggregationType as string
+  );
+  const alertDurationMS =
+    alert && alert.fields[ALERT_DURATION]
+      ? // duration is us, convert it to MS
+        (alert.fields[ALERT_DURATION] as unknown as number) / 1000
+      : 0;
```

Comment:
Why is there a fallback here? Is `ALERT_DURATION` ever nullish? If it's just a type issue, either the type probably should be fixed, or if that's too complicated, use the nullish coalescing operator (`!`).